### PR TITLE
feature (webapi): restricted view config

### LINKED ISSUE #690: Configure restricted view in project settings
I originally created this issue in Sipi repository [#224](https://github.com/dhlab-basel/Sipi/issues/224) as a reminder of a discussion with @lrosenth a few weeks ago.

Some project will need to **define the maximum resolution allowed** for their publicly available images. Due to some restrictions, some projects are allowed to display images : their resolution should be good enough for the unknown user to view but not good enough to make further image processing treatments interesting for this user. 

@lrosenth talked about a **configuration script for Sipi** per project. That would imply to create **a new property in Knora** to deal with this file in (I guess) admin data.

### PR DESCRIPTION
**Pull request by subotic**
### Summary

- add `kb:projectRestrictedViewSize` and `kb: projectRestrictedViewWatermark` optional properties to admin ontology
- add querying of the project's restricted view settings
- changing will be added in a future PR
- change Sipi preflight scripts to adhere to the supplied restricted view settings

### Issues
- resolves #690

### CODE REVIEW

**Comment by subotic** [comment]
@benjamingeer I hate Lua ;-) but I think it works now.

**Comment by benjamingeer** [comment]
Great, will try to review this on Monday.

File: webapi/src/test/scala/org/knora/webapi/responders/admin/SipiResponderADMSpec.scala [hunk 1]
```diff
@@ -20,43 +20,55 @@
...
 
 import scala.concurrent.duration._
 
-object SipiResponderADMSpec {
-
-    // A test UserProfileV1.
-    private val requestingUser: UserADM = SharedTestDataADM.incunabulaMemberUser
-
-    private val fileValueResponseFull = SipiFileInfoGetResponseADM(permissionCode = 6)
+object SipiResponderV1Spec {
```
Comment (benjamingeer) [inline webapi/src/test/scala/org/knora/webapi/responders/admin/SipiResponderADMSpec.scala]: This looks like a merge error. The responder is now `SipiResponderADM`, so the spec is called `SipiResponderADMSpec`.

File: docs/src/paradox/03-apis/api-admin/projects.md [hunk 2]
```diff
@@ -0,0 +1,144 @@
...
+  - Required permission: SystemAdmin / ProjectAdmin
+  - Required information: project IRI
+  - GET: `/admin/projects/members/<projectIri>`
+
+
+### Restricted View Settings Operations
+
+Operates on the following properties:
+ - `knora-base:projectRestrictedViewSize` - takes the IIIF size value
+ - `knora-base:projectRestrictedViewWatermark` - takes the path to the watermark image. **Currently not used.**
```
Comment (benjamingeer) [inline docs/src/paradox/03-apis/api-admin/projects.md]: Would it be possible to specify that I want restricted view to have:

* A size without a watermark
* A watermark without a size
* Both a size and a watermark

Comment (subotic) [reply docs/src/paradox/03-apis/api-admin/projects.md]: Currently, Sipi allows only one or the other. Also,  only size is implemented:

https://github.com/dhlab-basel/Knora/blob/663b1b1bdc3e244b694c621ecfc61a9eab93a096/sipi/config/sipi.init-knora.lua#L139-L141

As soon as Sipi allows both, then whatever combination is specified in the project, will also be returned by Sipi:

- only size
- only watermark
- both size and watermark

File: webapi/src/test/scala/org/knora/webapi/responders/admin/SipiResponderADMSpec.scala (hunk 1 above)
Comment (subotic) [reply webapi/src/test/scala/org/knora/webapi/responders/admin/SipiResponderADMSpec.scala]: changed

File: docs/src/paradox/03-apis/api-admin/projects.md (hunk 2 above)
Comment (benjamingeer) [reply docs/src/paradox/03-apis/api-admin/projects.md]: OK!

**Comment by loicjaouen** [comment]
thanks, it exactly implements  #842 and probably #86 ?

**Pull request merged**